sharding/contracts/sharding_manager.sol

```diff
@@ -1,253 +1,260 @@
+  uint constant NOTARY_REWARD = 0.001 ether;
+  // The minimum deposit size for a proposer
+  uint constant PROPOSER_DEPOSIT = 1 ether;
+  // The minimum balance of a proposer
+  uint constant MIN_PROPOSER_BALANCE = 0.1 ether;
+  // Time the ether is locked by notaries
+  uint NOTARY_LOCKUP_LENGTH = 16128;
+  // Time the ether is locked by proposers
+  uint PROPOSER_LOCKUP_LENGTH = 48;
   // Number of periods ahead of current period, which the contract
-  // is able to return the collator of that period
-  uint constant lookAheadPeriods = 4;
+  // is able to return the notary of that period
+  uint constant LOOKAHEAD_LENGTH = 4;
+  // Number of notaries to select from notary pool for each shard in each period
+  uint constant COMMITTEE_SIZE = 135;
+  // Threshold(number of notaries in committee) for a proposal to be deem accepted
+  uint constant QUORUM_SIZE = 90;
 
   // Log the latest period number of the shard
   mapping (int => int) public periodHead;
 
-  function SMC() public {
-  }
-
-  // Returns the gas limit that collations can currently have (by default make
-  // this function always answer 10 million).
-  function getCollationGasLimit() public pure returns(uint) {
-    return 10000000;
-  }
-
-  // Uses a block hash as a seed to pseudorandomly select a signer from the collator pool.
-  // [TODO] Chance of being selected should be proportional to the collator's deposit.
-  // Should be able to return a value for the current period or any future period up to.
-  function getEligibleCollator(int _shardId, uint _period) public view returns(address) {
-    require(_period >= lookAheadPeriods);
-    require((_period - lookAheadPeriods) * periodLength < block.number);
-    require(numCollators > 0);
-    // [TODO] Should check further if this safe or not
-    return collators[
-      int(
-        uint(
-          keccak256(
-            uint(block.blockhash((_period - lookAheadPeriods) * periodLength)),
-            _shardId
-          )
-        ) %
-        uint(getCollatorsMaxIndex())
-      )
-    ].addr;
-  }
-
-  function deposit() public payable returns(int) {
-    require(!isCollatorDeposited[msg.sender]);
-    require(msg.value == depositSize);
-    // Find the empty slot index in collators pool
-    int index;
-    if (!isStackEmpty())
+  function SMC(uint notary_lockup_length, uint proposer_lockup_length) public {
+    NOTARY_LOCKUP_LENGTH = notary_lockup_length;
+    PROPOSER_LOCKUP_LENGTH = proposer_lockup_length; 
+  }
+
+  event LOG(uint L);
+
+
+  function isNotaryInCommittee(uint shardId, uint period) public view returns(bool) {
+    uint currentPeriod = block.number / PERIOD_LENGTH;
+
+    // Determine notary pool length based on notary sample size
+    uint sampleSize;
+    if (period > sampleSizeLastUpdatedPeriod) {
+      sampleSize = nextPeriodNotarySampleSize;
+    } else {
+      sampleSize = currentPeriodNotarySampleSize;
+    }
+
+    uint periodToLook = period * PERIOD_LENGTH - 1;
```

Comment:
What is this? Looks like you convert the period to a block number then subtract one? 
But the variable is called "periodToLook". Consider renaming this so it is easier to understand. 

Also consider adding a comment to explain where you are looking "Check the most recent block before the period started" or something like that.

Reply:
Ah! good catch! it should be block